# List of papers about Proteins Design using Deep Learning

## About this repository

Inspired by Kevin Kaichuang Yang's [Machine-learning-for-proteins](https://github.com/yangkky/Machine-learning-for-proteins). In terms of the fast changing of protein design in DL, the dynamic repository will be a record of papers for these newcomers like me.  

## Menu

* Hallucination
* function to scaffold
* scaffold to sequence
* function to sequence
* Generative Models

## Hallucination

### trRosetta/trDesign-based

 **Design of proteins presenting discontinuous functional sites using deep learning**  
Tischer, Doug, et al.  
[bioRxiv (2020)](https://www.biorxiv.org/content/10.1101/2020.11.29.402743v1.abstract)  

**De novo protein design by deep network hallucination**  
Anishchenko, I., Pellock, S.J., Chidyausiku, T.M. et al.  
[Nature (2021)](https://doi.org/10.1038/s41586-021-04184-w)  

**Structure-based protein design with deep learning**  
Ovchinnikov, Sergey, and Po-Ssu Huang.  
[Current opinion in chemical biology 65 (2021): 136-144.](https://www.sciencedirect.com/science/article/pii/S1367593121001125)  

**Protein sequence design by conformational landscape optimization**
Norn, Christoffer, et al.  
[Proceedings of the National Academy of Sciences 118.11 (2021)](https://www.pnas.org/content/118/11/e2017228118)

### AlphaFold2-based

**End-to-end learning of multiple sequence alignments with differentiable Smith-Waterman**  
Petti, Samantha, et al.  
[bioRxiv (2021)](http://repository.cshl.edu/id/eprint/40409/)  

**AlphaDesign: A de novo protein design framework based on AlphaFold**  
Jendrusch, Michael, Jan O. Korbel, and S. Kashif Sadiq.  
[bioRxiv (2021)](https://www.biorxiv.org/content/10.1101/2021.10.11.463937.abstract)  

## Function to scaffold

**Design of metalloproteins and novel protein folds using variational autoencoders**
Greener, Joe G., Lewis Moffat, and David T. Jones.  
[Scientific reports 8.1 (2018)](https://www.nature.com/articles/s41598-018-34533-1)  

**Function-guided protein design by deep manifold sampling**  
Vladimir Gligorijevic, Stephen Ra, Daniel Berenberg, Richard Bonneau, Kyunghyun Cho  
[NeurIPS2021](https://www.mlsb.io/papers_2021/MLSB2021_Function-guided_protein_design_by.pdf)

## Scaffold/Template to sequence

**DenseCPD: improving the accuracy of neural-network-based computational protein sequence design with DenseNet**
Qi, Yifei, and John ZH Zhang.  
[Journal of chemical information and modeling 60.3 (2020)](https://pubs.acs.org/doi/pdf/10.1021/acs.jcim.0c00043)

**TERMinator: A Neural Framework for Structure-Based Protein Design using Tertiary Repeating Motifs**
Li, Alex J., et al.  
[NeurIPS 2021](https://www.mlsb.io/papers_2021/MLSB2021_TERMinator:_A_Neural_Framework.pdf)

**Fast and flexible protein design using deep graph neural networks.**
Strokach, Alexey, et al.  
[Cell Systems 11.4 (2020)](https://www.sciencedirect.com/science/article/pii/S2405471220303276)

**Computational generation of proteins with predetermined three-dimensional shapes using ProteinSolver.**
Strokach, Alexey, et al.  
[STAR protocols 2.2 (2021)](https://www.sciencedirect.com/science/article/pii/S2666166721002124)

**Deep learning methods for designing proteins scaffolding functional sites**  
Wang, J., et al.  
[bioRxiv(2021)](https://europepmc.org/article/ppr/ppr419387)  

**Computational protein design with deep learning neural networks**  
Wang, Jingxue, et al.  
[Scientific reports 8.1 (2018)](https://www.nature.com/articles/s41598-018-24760-x.pdf)

## Function to sequence

**Deep neural language modeling enables functional protein generation across families**  
Madani, Ali, et al.  
[bioRxiv (2021)](https://www.biorxiv.org/content/10.1101/2021.07.18.452833.abstract)  

## Generative Models

### CM-Align

**AutoFoldFinder: An Automated Adaptive Optimization Toolkit for De Novo Protein Fold Design**  
Shuhao Zhang, Youjun Xu, Jianfeng Pei, Luhua Lai
[NeurIPS2021](https://www.mlsb.io/papers_2021/MLSB2021_AutoFoldFinder.pdf)  

### GAN

**Conditional Generative Modeling for De Novo Protein Design with Hierarchical Functions**  
Kucera, Tim, Matteo Togninalli, and Laetitia Meng-Papaxanthos  
[bioRxiv (2021)](https://www.biorxiv.org/content/10.1101/2021.11.10.467885v1.abstract)  

**Expanding functional protein sequence spaces using generative adversarial networks**
Repecka, Donatas, et al.  
[Nature Machine Intelligence 3.4 (2021)](https://www.nature.com/articles/s42256-021-00310-5)

### NLP-based

**Recurrent neural network model for constructive peptide design**  
MuÌˆller, Alex T., Jan A. Hiss, and Gisbert Schneider.  
[Journal of chemical information and modeling 58.2 (2018)](https://pubs.acs.org/doi/pdf/10.1021/acs.jcim.7b00414)  

### GNN-based

Generative models for graph-based protein design
Ingraham, John, et al.  
[NeurIPS2019](https://openreview.net/forum?id=ByMEAHrgLB)

### Pretrained-based

**Progen: Language modeling for protein generation**
Madani, Ali, et al.  
[arXiv preprint arXiv:2004.03497 (2020)](https://arxiv.org/abs/2004.03497)

## Reviews
**Deep generative modeling for protein design.**
Strokach, Alexey, and Philip M. Kim.  
[Current Opinion in Structural Biology 72 (2022)](https://www.sciencedirect.com/science/article/pii/S0959440X21001573)